pkg/render/kubecontrollers/kube-controllers.go

```diff
@@ -244,22 +269,47 @@ func (c *kubeControllersComponent) controllersRole() *rbacv1.ClusterRole {
 				Resources: []string{"licensekeys"},
 				Verbs:     []string{"get"},
 			},
-			{
-				APIGroups: []string{"crd.projectcalico.org"},
-				Resources: []string{"remoteclusterconfigurations"},
-				Verbs:     []string{"watch", "list", "get"},
-			},
-			{
-				// For federated services.
-				APIGroups: []string{""},
-				Resources: []string{"endpoints"},
-				Verbs:     []string{"create", "update", "delete"},
-			},
-			{
-				APIGroups: []string{"rbac.authorization.k8s.io"},
-				Resources: []string{"clusterroles", "clusterrolebindings"},
-				Verbs:     []string{"watch", "list", "get"},
-			},
+		}
+
+		if c.kubeControllerName == KubeController {
+			extraRules = append(extraRules,
```

Comment:
Can you make this decision in the constructors as well please, and set them to a rules variable. Then you can just render the rules outright.